### ISSUE TITLE
react/cache 0.4 requirement creating problems

### ISSUE THREAD

**Issue by esokullu** [WyriHaximus/reactphp-cache-filesystem#22]
With newer packages that require react/cache 0.5

**Comment by WyriHaximus** [comment]
It's on my todo list for a while now. I'll try and get an update out later this week

**Comment by WyriHaximus** [comment]
Whoops seems this one fell through. If you're up to it you can PR it, recently did another for 1.0, see https://github.com/WyriHaximus/reactphp-cache-redis/blob/master/src/Redis.php for inspiration. Not sure when I would have time for it as this one is pretty low on my priority list.

**Comment by WyriHaximus** [comment]
@esokullu just pushed 60587f2d2b4b367692b6097430791ed2c2c0b0b0 which includes all the changes to work with the latest cache and filesystem components

**Comment by esokullu** [comment]
Sorry, I just saw these!

I was trying to give it a try, but here's the error message I receive:

```
PHP Warning:  strlen() expects parameter 1 to be string, array given in [....]/vendor/react/filesystem/src/Stream/WritableStreamTrait.php on line 14
PHP Warning:  base64_encode() expects parameter 1 to be string, array given in [...]/vendor/react/filesystem/src/ChildProcess/Adapter.php on line 216
```

with a setup like:

```
$filesystem = ReactFilesystem::create($this->loop);
$cache = new Filesystem($filesystem, sys_get_temp_dir());
```

This may be a React/Filesystem issue as well. Meanwhile, I am switching to the EIO extension to see if that is what's causing the issue.

**Comment by esokullu** [comment]
yes, works like a breeze with eio installed. So, must be a React\Filesystem issue.

**Comment by WyriHaximus** [comment]
Interesting, does react/filesystem work when you have the EIO extension installed or are you using EIO directly?

**Comment by esokullu** [comment]
No I wasn't using EIO directly. But there are still some issues. I'm investigating, I'll let you know.

**Comment by esokullu** [comment]
Aargh... I was wrong, I'm still getting this same issue: 
```
PHP Warning:  strlen() expects parameter 1 to be string, array given in [...]/vendor/react/filesystem/src/Stream/WritableStreamTrait.php on line 14
PHP Notice:  Array to string conversion in [...]/vendor/react/filesystem/src/Eio/Adapter.php on line 410
```

(slightly edited to hide paths)

**Comment by WyriHaximus** [comment]
> Aargh... I was wrong, I'm still getting this same issue:
> 
> ```
> PHP Warning:  strlen() expects parameter 1 to be string, array given in [...]/vendor/react/filesystem/src/Stream/WritableStreamTrait.php on line 14
> PHP Notice:  Array to string conversion in [...]/vendor/react/filesystem/src/Eio/Adapter.php on line 410
> ```

Could you try `composer require react/filesystem:"dev-master as v0.1.13"`?

> (slightly edited to hide paths)

Hah I do the same 😄
